elastic/client.go

```diff
@@ -214,7 +228,7 @@ func (c *Client) CreateMapping(index string, docType string, mapping map[string]
 		return errors.Trace(err)
 	}
 
-	// index doesn't exist, create index first
+	// if index doesn't exist, will get 404 not found error, create index first
 	if r.Code != http.StatusOK {
```

Comment:
maybe we can check 404 here directly and return an error for other invalid statuses.